Source/WebCore/page/scrolling/ScrollingCoordinatorTypes.cpp

```diff
@@ -34,7 +34,7 @@ void RequestedScrollData::merge(RequestedScrollData&& other)
         // Carry over the previously requested scroll position so that we can set `requestedDataBeforeAnimatedScroll`
         // below in the case where the cancelled animated scroll is immediately followed by another animated scroll.
         other.scrollPosition = scrollPosition;
-    } else if (other.requestType == ScrollRequestType::PositionUpdate && other.animated == ScrollIsAnimated::Yes) {
+    } else if ((other.requestType == ScrollRequestType::PositionUpdate || other.requestType == ScrollRequestType::DeltaUpdate) && other.animated == ScrollIsAnimated::Yes) {
```

Comment:
I'm not sure this merging is correct. If there's a requested ScrollRequestType::PositionUpdate already pending, and a DeltaUpdate comes in, should you add the delta to the position request? Can you make a test that exercises this code?

Reply:
I don't think it's correct to add a DeltaUpdate to a PositionUpdate. These two should be functionally equivalent to each other, so they should use the same existing merge behavior, where a new animated scroll overwrites the previous animated scroll (and passes forward the `requestedDataBeforeAnimatedScroll` if it exists), and a non animated scroll simply overwrites the current RequestedScrollData. This merge code is pretty well covered by current tests (like those in imported/w3c/web-platform-tests/css/cssom-view/scroll-behavior-*).